packages/formatters/src/formatters.ts

```diff
@@ -373,6 +373,92 @@ export function messageLink<C extends Snowflake, M extends Snowflake, G extends
 	return `${guildId === undefined ? channelLink(channelId) : channelLink(channelId, guildId)}/${messageId}`;
 }
 
+export enum HeadingLevel {
+	One = 1,
+	Two,
+	Three,
+}
+
+/**
+ * Formats the content into a heading level.
+ *
+ * @typeParam C - This is inferred by the supplied content
+ * @param content - The content to wrap
+ * @param level - The heading level
+ */
+export function heading<C extends string>(content: C, level?: HeadingLevel.One): `# ${C}`;
+
+/**
+ * Formats the content into a heading level.
+ *
+ * @typeParam C - This is inferred by the supplied content
+ * @param content - The content to wrap
+ * @param level - The heading level
+ */
+export function heading<C extends string>(content: C, level: HeadingLevel.Two): `## ${C}`;
+
+/**
+ * Formats the content into a heading level.
+ *
+ * @typeParam C - This is inferred by the supplied content
+ * @param content - The content to wrap
+ * @param level - The heading level
+ */
+export function heading<C extends string>(content: C, level: HeadingLevel.Three): `### ${C}`;
+
+export function heading(content: string, level?: HeadingLevel) {
+	switch (level) {
+		case HeadingLevel.Three:
+			return `### ${content}`;
+		case HeadingLevel.Two:
+			return `## ${content} `;
+		default:
+			return `# ${content}`;
+	}
+}
+
+export type RecursiveArray<T> = readonly (RecursiveArray<T> | T)[];
+
+/**
+ * Formats the elements in the array to an ordered list.
+ *
+ * @param list - The array of elements to list
+ * @param startNumber - The starting number for the list.
+ */
+export function orderedList(list: RecursiveArray<string>, startNumber: number = 1): string {
+	let newStartNumber = startNumber;
+	if (startNumber < 0) newStartNumber = 1;
+
+	const processElement = (element: RecursiveArray<string>, depth: number = 0): string => {
+		if (Array.isArray(element)) {
+			const results = element.map((element) => processElement(element, depth + 1));
+			return results.join('\n');
+		}
+
+		return `${'  '.repeat(depth - 1)}${newStartNumber}. ${element}`;
+	};
```

Comment:
This seems almost identical to the one in `unorderedList()`. Perhaps it can be brought to the outer scope, marked as `@internal`, and deduplicated?

Reply:
I was going to put it in a `util` file or folder but formatters doesn't have something like that, can you please explain what you mean by marked as `@internal`?

Reply:
You could probably create a util folder then.

[`@internal`](https://tsdoc.org/pages/tags/internal) from TSDoc marks the symbol for internal usage only. Example:

https://github.com/discordjs/discord.js/blob/7295a3a94a853086f781e7810b37c77d0c7e5283/packages/rest/src/lib/handlers/Shared.ts#L20-L25

It won't appear on the documentation. It's like JSDoc's [`@private`](https://jsdoc.app/tags-private.html).

Reply:
Alright, sounds good, and thank you for the explanation, I understand now ❤️